src/integrationTest/java/org/opensearch/security/IpBruteForceAttacksPreventionTests.java

```diff
@@ -0,0 +1,149 @@
+import org.opensearch.test.framework.RateLimiting;
+import org.opensearch.test.framework.TestSecurityConfig.User;
+import org.opensearch.test.framework.cluster.ClusterManager;
+import org.opensearch.test.framework.cluster.LocalCluster;
+import org.opensearch.test.framework.cluster.TestRestClient;
+import org.opensearch.test.framework.cluster.TestRestClient.HttpResponse;
+import org.opensearch.test.framework.cluster.TestRestClientConfiguration;
+
+import static org.apache.hc.core5.http.HttpStatus.SC_OK;
+import static org.apache.hc.core5.http.HttpStatus.SC_UNAUTHORIZED;
+import static org.opensearch.test.framework.TestSecurityConfig.AuthcDomain.AUTHC_HTTPBASIC_INTERNAL_WITHOUT_CHALLENGE;
+import static org.opensearch.test.framework.TestSecurityConfig.Role.ALL_ACCESS;
+import static org.opensearch.test.framework.cluster.TestRestClientConfiguration.userWithSourceIp;
+
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class IpBruteForceAttacksPreventionTests {
+	private static final User USER_1 = new User("simple-user-1").roles(ALL_ACCESS);
+	private static final User USER_2 = new User("simple-user-2").roles(ALL_ACCESS);
+
+	public static final int ALLOWED_TRIES = 3;
+	public static final int TIME_WINDOW_SECONDS = 3;
+
+	public static final String CLIENT_IP_2 = "127.0.0.2";
+	public static final String CLIENT_IP_3 = "127.0.0.3";
+	public static final String CLIENT_IP_4 = "127.0.0.4";
+	public static final String CLIENT_IP_5 = "127.0.0.5";
+	public static final String CLIENT_IP_6 = "127.0.0.6";
+	public static final String CLIENT_IP_7 = "127.0.0.7";
+	public static final String CLIENT_IP_8 = "127.0.0.8";
+	public static final String CLIENT_IP_9 = "127.0.0.9";
+
+	private static final AuthFailureListeners listener = new AuthFailureListeners()
+		.addRateLimit(new RateLimiting("internal_authentication_backend_limiting").type("ip")
+			.allowedTries(ALLOWED_TRIES).timeWindowSeconds(TIME_WINDOW_SECONDS).blockExpirySeconds(2).maxBlockedClients(500)
+			.maxTrackedClients(500));
+
+	@ClassRule
+	public static final LocalCluster cluster = new LocalCluster.Builder()
+		.clusterManager(ClusterManager.SINGLENODE).anonymousAuth(false).authFailureListeners(listener)
+		.authc(AUTHC_HTTPBASIC_INTERNAL_WITHOUT_CHALLENGE).users(USER_1, USER_2).build();
+
+	@Test
+	public void shouldAuthenticateUserWhenBlockadeIsNotActive() {
+		try(TestRestClient client = cluster.createGenericClientRestClient(userWithSourceIp(USER_1, CLIENT_IP_2))) {
+
+			HttpResponse response = client.getAuthInfo();
+
+			response.assertStatusCode(SC_OK);
+		}
+	}
+
+	@Test
+	public void shouldBlockIpAddress() {
+		authenticateUserWithIncorrectPassword(CLIENT_IP_3, USER_2, ALLOWED_TRIES);
+		try(TestRestClient client = cluster.createGenericClientRestClient(userWithSourceIp(USER_2, CLIENT_IP_3))) {
+
+			HttpResponse response = client.getAuthInfo();
+
+			response.assertStatusCode(SC_UNAUTHORIZED);
```

Comment:
Out of curiosity, how do you test that an IP is blocked after multiple failed logins